packages/@react-aria/calendar/src/useCalendarCell.ts

```diff
@@ -101,7 +101,11 @@ export function useCalendarCell(props: AriaCalendarCellProps, state: CalendarSta
   );
 
   if (isInvalid) {
-    isSelected = true;
+    // don't mark unavaliable + invalid dates in range calendars as selected, that case only comes up via allowsNoncontiguousRanges
+    // and thus those unavailable dates shouldn't be selected. For single select calendars, we mark unavailble + invalid days as selected for styling reasons
+    if (!('highlightedRange' in state) || !isUnavailable) {
+      isSelected = true;
+    }
```

Comment:
behavior open to debate, but the single select calendar case had a prior test for it hence why the special casing. I think this makes sense, range calendars don't actually include unavailable days that fall inside their range selection, but a single select calendar might have a value/default value set that is invalid + unavailable